src/policy.ts

```diff
@@ -107,7 +107,19 @@ export class MultisigWalletPolicy {
     template: string;
     keyOrigins: KeyOrigin[];
   }) {
-    this.name = name;
+    // this is an unstated restriction from ledger
+    // if we don't check it here then registration will fail
+    // with an opaque error about invalid input data
+    // TODO: should this be left as full length and only
+    // abbreviated when translating to a ledger policy?
+    if (name.length > 64) {
```

Comment:
Maybe keep an eye on this. I saw it reporting name as undefined in the test suite, but things might not be aligning there.

Reply:
I think this was just because the fixtures in your environment weren't up to date so an undefined variable was getting passed to the interaction class. If caravan were in typescript too that would've failed during build.